src/components/instantlaunches/index.js

```diff
@@ -0,0 +1,176 @@
+                inputParamTypes.includes(param.type)
+            );
+
+            // Get listing of the input parameters from the app info that
+            // aren't set in the QL submission
+            const unsetInputParams = appInputs.filter((appParam) => {
+                return (
+                    !submission.config.hasOwnProperty(appParam.id) ||
+                    (Array.isArray(submission.config[appParam.id]) &&
+                        submission.config[appParam.id].length === 0)
+                );
+            });
+
+            // For each unset input parameter, match it up with an appropriate
+            // resource that was passed in to the function. File resources should
+            // go with FileInput params or MultiFileSelectors, while folder resources
+            // should go with FolderInput params.
+            if (unsetInputParams.length > 0) {
+                for (const unsetParam of unsetInputParams) {
+                    if (resource.type === "file") {
+                        if (
+                            unsetParam.type ===
+                            launchConstants.PARAM_TYPE.FILE_INPUT
+                        ) {
+                            submission.config[unsetParam.id] = resource.path;
+                            break;
+                        }
+
+                        if (
+                            unsetParam.type ===
+                            launchConstants.PARAM_TYPE.MULTIFILE_SELECTOR
+                        ) {
+                            submission.config[unsetParam.id] = [resource.path];
+                            break;
+                        }
+                    }
+
+                    if (resource.type === "folder") {
+                        if (
+                            unsetParam.type ===
+                            launchConstants.PARAM_TYPE.FOLDER_INPUT
+                        ) {
+                            submission.config[unsetParam.id] = resource.path;
+                            break;
+                        }
+                    }
+                }
+            }
+            return submission;
+        })
+        .then((submission) => submitAnalysis(submission)) // submit the analysis
+        .then((analysisResp) => getAnalysis(analysisResp.id));
+};
+
+/**
+ * An IconButton with a handler that launches the app in the instant launch with
+ * the resource as an input.
+ *
+ * @param {Object} props
+ * @param {Object} props.instantLaunch - The instant launch to use.
+ * @param {Object} props.resource - The resource to use as an input to the instant launch.
+ */
+const InstantLaunchButton = ({
+    instantLaunch,
+    resource,
+    showErrorAnnouncer,
+}) => {
+    const { t } = useTranslation(["instantlaunches", "common"]);
+    return (
+        <IconButton
+            variant="contained"
+            onClick={async (event) => {
+                event.stopPropagation();
+                event.preventDefault();
```

Comment:
The above two lines prevent a click on the instant launch button from selecting the resource row in the data listing. That got a little confusing and could imply that we support sending multiple files to a single instantly launched app, which isn't the case (yet).